crates/starknet-server/src/api/models/transaction.rs

```diff
@@ -143,46 +143,27 @@ pub struct L1HandlerTransaction {
     pub calldata: Calldata,
 }
 
-/// A transaction status in Starknet.
-#[derive(Debug, Copy, Clone, Eq, PartialEq, Deserialize, Serialize, Default)]
-pub enum TransactionStatus {
-    /// The transaction passed the validation and entered the pending block.
-    #[serde(rename = "PENDING")]
-    Pending,
-    /// The transaction passed the validation and entered an actual created block.
-    #[serde(rename = "ACCEPTED_ON_L2")]
-    #[default]
-    AcceptedOnL2,
-    /// The transaction was accepted on-chain.
-    #[serde(rename = "ACCEPTED_ON_L1")]
-    AcceptedOnL1,
-    /// The transaction failed validation.
-    #[serde(rename = "REJECTED")]
-    Rejected,
-}
-
-#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[derive(Debug, Clone, Eq, PartialEq, Serialize)]
 pub struct TransactionReceiptWithStatus {
     pub status: TransactionStatus,
     #[serde(flatten)]
     pub receipt: TransactionReceipt,
 }
 
-#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[derive(Debug, Clone, Eq, PartialEq, Serialize)]
 #[serde(untagged)]
 pub enum TransactionReceipt {
-    Deploy(DeployTransactionReceipt),
     Common(CommonTransactionReceipt),
 }
 
-#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[derive(Debug, Clone, Eq, PartialEq, Serialize)]
 pub struct DeployTransactionReceipt {
     #[serde(flatten)]
     pub common: CommonTransactionReceipt,
     pub contract_address: ContractAddress,
 }
 
-#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
```

Comment:
Why are deserialisations removed?

Reply:
because it was used only in serialization

Reply:
We have dump/load feature that will be affected by this

Reply:
all these types are used in responses, not in inputs

Reply:
okay, didn't noted about the dump/load feature

Reply:
fixed